src/anomalib/trainer/callbacks/normalization/cdf_normalization.py

```diff
@@ -104,16 +103,16 @@ def on_predict_batch_end(
         self._normalize_batch(outputs, pl_module)
         outputs["pred_labels"] = outputs["pred_scores"] >= 0.5
 
-    def _collect_stats(self, trainer: pl.Trainer, pl_module: AnomalyModule) -> None:
+    def _collect_stats(self, trainer: "anomalib.AnomalibTrainer", pl_module: AnomalyModule) -> None:
         """Collect the statistics of the normal training data.
 
         Create a trainer and use it to predict the anomaly maps and scores of the normal training data. Then
          estimate the distribution of anomaly scores for normal data at the image and pixel level by computing
          the mean and standard deviations. A dictionary containing the computed statistics is stored in self.stats.
         """
-        predictions = AnomalibTrainer(accelerator=trainer.accelerator, devices=trainer.num_devices).predict(
-            model=self._create_inference_model(pl_module), dataloaders=trainer.datamodule.train_dataloader()
-        )
+        predictions = anomalib.AnomalibTrainer(
```

Comment:
1. It looks a bit weird with `anomalib.AnomalibTrainer` just to avoid circular import. I'll try to find a better solution for this.
2. I guess the idea is to get normalization statistics from the training set instead of the validation set. Dick is probably the best person to address this but I did notice some difference in scores when I used the validation data. Plus the number of images in validation set might not be large enough to compute a reasonable statistics. We can't move gathering the stats to the training step as a trained model is not available yet to produce a decent anomaly map. But maybe the whole normalization requires refactor. 
3. 👍